### ISSUE TITLE
Server Setup Incomplete

### ISSUE THREAD

**Issue by CalMatar** [exodus4d/pathfinder#873]
I have spent the last several days trying to get Pathfinder set up on my server and finally had someone else assist me with the configuration and SSL setup and I kept getting errors. The problems I kept running into are the fact that there is code that is not included in the get that is referenced in the Pathfinder.conf file for the SSL such as all the stuff related to h5bp. There is nothing in the get related to this and I can not find it anywhere even though I have downloaded the Nginx system from them. 
Any help on this issue would be greatly appreciated. Also making documentation for anyone that does not have Pathfinder installed already would be great too as this is really hard to follow even for people that use Linux and Ubuntu all the time.

**Comment by porowns** [comment]
Just use docker and proxypass, it's infinitely easier. https://github.com/KryptedGaming/pathfinder-docker

**Comment by eophten** [comment]
As an alternative to docker and proxypass, the setup is fairly simple on a hosted server in AWS with EC2 and an SSL enabled ELB

**Comment by november781** [comment]
seconding the project @porowns linked, I'm using it 24/7 for my own instillation. Having no issues personally, tossed it up on a nanode from linode and it is running really well.

**Comment by exodus4d** [comment]
@CalMatar  The Nginx config files from h5bp are not required for SSL. You can remove the include from your `pathfinder.ini`:
```
include h5bp/basic.conf;
```
GitHub repo: https://github.com/h5bp/server-configs-nginx

**Comment by CalMatar** [comment]
Thank you @exodus4d.That is a big help as I think that was where my biggest problem was. I just redid my server due to some issues I had so I am going to be reinstalling everything and will do this again and now that I know what I can do without/change I am hoping to get it up and running.
I will keep you updated.

**Comment by CalMatar** [comment]
Ok, I have been wracking my brain and have been trying to find the file listed in your Pathfinder.conf file and can not find it anywhere in any of the gits at all. I am not sure if I have not done something that I was supposed to do so far but am I missing something? 
The part of the file that I am talking about is this:
  # Include the basic SSL h5bp config set
  include h5bp/directive-only/ssl.conf;
That folder and file do not exist in the h5bp git. I can not figure out if I am not doing something I should have or doing something I shouldn't have or I should just delete those lines altogether. If I do that then there is actually no reference to h5bp at all in the config file for pathfinder and I am not sure what the purpose of using his setup is. Can you please explain that to me please? I am a bit hazy on the reasons why a standard Nginx configuration for multiple domains, which I have already painstakingly set up, will not work for this. I am having a hard time trying to figure out where the h5bp setup is more beneficial as I have, seemingly, everything already done that his setup redid but I would like for you to explain your reasoning. 
Much appreciated thank you.

**Comment by exodus4d** [comment]
1. If you recently deleted the include of any *.conf file or changed any *.conf file, you **need to** either **restart** Nginx or better **reload** Nginx config. Unix: `nginx -r`.
2. Make sure that ther is also no includ of any `/h5bp/.../*.conf` in the **main** `nginx.conf` file, too, in order to remove any include of h5bp. 
This is the primary config file which is **always** used by Nginx and includes all the domain specific *.conf files like `pathfinder.conf` at the end:
   ```
   include domains.d/*.conf;
   ```

**Comment by CalMatar** [comment]
Right. That is already done but that is not the problem. As previously stated the issue is not with Nginx directly it is with the pathfinder.conf file. In your pathfinder.conf file you have a line that references a file that does not exist in the h5bp setup anywhere. It deals with the ssl setup and for the life of me I can not seem to find it, or figure out how to produce it, other than simply create it out of whole cloth, which I am loath to do. My thought was to simply comment out or delete the lines but I wanted to make sure that was not going to mess anything up if I did that. I have the SSL set up before that already so not sure what this did. 
Also, if I do delete this line what exactly is the h5bp setup being used for?

**Comment by exodus4d** [comment]
The intention was to have some config files available that can be included _on demand_ that handle some basic settings like cache control for all kinds of images, fonts,.... I used h5bp files in _PF_, PhpMyAdmin, .... and all other enabled sites in Nginx. E.g. : 
- https://github.com/h5bp/server-configs-nginx/blob/master/h5bp/web_performance/cache_expiration.conf handles default HTTP Cache headers for static resources. Then If I want to change/overwrite those default settings, I can do this in my `pathfinder.conf` `location{...}` section.
- https://github.com/h5bp/server-configs-nginx/blob/master/h5bp/location/security_file_access.conf prevents access to system files like `*.bak`, `*.ini`, `.sh`,... This is just for more secirity. If you accidentally move your `app/environment.ini` file (which has all the DB credentials in it) into your `public/...` dir, Nginx deny access....

**Comment by CalMatar** [comment]
Ok, I can try it again, but the problem I listed above still will exist. What is to be done about the directive-only folder that does not exist in the h5bp system of folders? I can not find it and following everything that it tells me to do there is no where that it was created. 
I can not find the ssl.conf file anywhere in the h5bp git either and no where does there seem to be any explanation of either creating one or doing anything that should be creating one. 
This is where I have been running into a problem.
